### ISSUE TITLE
verbose_file_reads suggestion isn't applicable

### ISSUE THREAD

**Issue by smmalis37** [rust-lang/rust-clippy#9780]
### Summary

The provided reproducer has two issues that prevent the suggestion from being applicable:
1. The file is opened in a different function, meaning we don't have the filename in a local scope, and so can't pass it to fs::read.
2. We are calling seek on the file before reading, using fs::read would make this impossible.

### Lint Name

verbose_file_reads

### Reproducer

I tried this code:

```rust
use std::fs::File;
use std::io::Read;
use std::io::Seek;
use std::io::SeekFrom;

fn foo(mut file: &File) {
    let mut file_contents = Vec::new();
    file.seek(SeekFrom::Start(10)).unwrap();
    file.read_to_end(&mut file_contents).unwrap();
}

fn main() {
    foo(&File::open("foo.txt").unwrap());
}
```

I saw this happen:

```
warning: use of `File::read_to_end`
 --> src\main.rs:9:5
  |
9 |     file.read_to_end(&mut file_contents).unwrap();
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: requested on the command line with `-W clippy::verbose-file-reads`
  = help: consider using `fs::read` instead
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#verbose_file_reads
```

I expected to see this happen:
No warnings, or at least a usable suggestion.

### Version

```text
rustc 1.65.0 (897e37553 2022-11-02)
binary: rustc
commit-hash: 897e37553bba8b42751c67658967889d11ecd120
commit-date: 2022-11-02
host: x86_64-pc-windows-msvc
release: 1.65.0
LLVM version: 15.0.0
```
